ros/2/src/openvslam/src/openvslam_ros.cc

```diff
@@ -53,15 +53,50 @@ mono::mono(const std::shared_ptr<openvslam::config>& cfg, const std::string& voc
         node_.get(), "camera/image_raw", [this](const sensor_msgs::msg::Image::ConstSharedPtr& msg) { callback(msg); }, "raw", custom_qos_);
 }
 void mono::callback(const sensor_msgs::msg::Image::ConstSharedPtr& msg) {
-    const auto tp_1 = std::chrono::steady_clock::now();
-    const auto timestamp = std::chrono::duration_cast<std::chrono::duration<double>>(tp_1 - tp_0_).count();
-
+    const rclcpp::Time tp_1 = node_->now();
+    const double timestamp = (tp_1 - tp_0_).nanoseconds();
+    
     // input the current frame and estimate the camera pose
     SLAM_.feed_monocular_frame(cv_bridge::toCvShare(msg)->image, timestamp, mask_);
 
-    const auto tp_2 = std::chrono::steady_clock::now();
+    const rclcpp::Time tp_2 = node_->now();
+    const double track_time = (tp_2 - tp_1).nanoseconds()/to_sec;
```

Comment:
`(tp_2 - tp_1).seconds()` exists now! No need for the `to_sec` duration object

Reply:
I thought it will take only the integer part like when using with <chrono::seconds>. I'll change it then!